### PR TITLE
NXP-30192: always check Read permission in DocumentModelResolver

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

**Review by kevinleturc — DISMISSED** [review]

File: modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java [hunk 1]
```diff
@@ -197,6 +199,14 @@ public Object fetch(Object value, Object context) {
         return docHolder.getValue();
     }
 
+    protected boolean hasReadPermission(CoreSession session, DocumentRef docRef) {
```
Comment (efge) [inline modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java:202]: Maybe add a comment explaining why we don't use `session.exists` here.

File: modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/resolver/TestDocumentModelResolver.java [hunk 2]
```diff
@@ -671,6 +681,32 @@ public void testUserWithNoAccessToDocument() {
...
+        acp.addACE(ACL.LOCAL_ACL, new ACE("bob", "Read", true));
+        fooDoc.setACP(acp, true);
+        transactionalFeature.nextTransaction();
+
+        CoreSession bobSession = coreFeature.getCoreSession("bob");
+        DocumentModel resolvedDoc = (DocumentModel) fooDoc.getProperty(REPO_AND_ID_XPATH)
+                                                          .getObjectResolver()
+                                                          .fetch(bobSession);
+        // bob cannot read the referenced document
+        assertNull(resolvedDoc);
```
Comment (efge) [inline modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/resolver/TestDocumentModelResolver.java:707]: Could you also test explicitly what happens on a list? I guess here the list will contain a `null`, but as we discussed another option would be to skip the value from the list completely.

**Review by kevinleturc — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]

**Review by efge — APPROVED** [review]
